metadata-transformer/src/main/java/no/unit/nva/metadata/MetadataConverter.java

```diff
@@ -43,19 +48,11 @@ public MetadataConverter(Model metadata, String jsonld) {
 
     public CreatePublicationRequest toRequest() throws JsonProcessingException {
         CreatePublicationRequest request = objectMapper.readValue(jsonld, CreatePublicationRequest.class);
-
-        addDoi(request, metadata);
-
+        getDoiFromMetadata(metadata).ifPresent(request::addReferenceDoi);
```

Comment:
I think as the metadata here could become a longer list that refactoring each task in a method would be increase readability. 
e.g.

```
var request = newCreateRequestFromJsonLd() ; // or var request = newCreateRequestFromRdfData()
addDoi(request)
addLexvoLanguageCode(request)
```
tells immediately and very clearly what the function does. Although the code seems to simple to think about, the cognitive load accumulates when these things are around. If one method is used by only one another method, that is not a problem. It is more important that each method contains one level of abstraction and it is readable.

Reply:
I disagree; 

One-liner lambdas or functional-style method-references are not more unclear or a higher cognitive load than a one-line traditional method reference, it's just a matter of getting used to them. What _does_ in my opinion increase cognitive load is having to jump around in needlessly long code trying to follow the logic of an ostensibly simple function. 

Your reasoning is otherwise sound, but you're optimizing for a potential problem that does not exist (yet?). That's premature.